rust/scripts/ci/generate-workspace-changelog.sh

```
@@ -0,0 +1,251 @@
+fi
+
+# Parse workspace filter into array
+if [ -n "$WORKSPACE_FILTER" ]; then
+    IFS=',' read -ra FILTER_ARRAY <<< "$WORKSPACE_FILTER"
+else
+    FILTER_ARRAY=()
+fi
+
+# Temporary directory for categorization
+TEMP_DIR=$(mktemp -d)
+trap "rm -rf $TEMP_DIR" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)